app/src/main/kotlin/edu/stanford/bdh/engagehf/MainActivity.kt

```diff
@@ -53,6 +65,10 @@ class MainActivity : FragmentActivity() {
 
     private val viewModel by viewModels<MainActivityViewModel>()
 
+    private val bluetoothViewModel by viewModels<BluetoothViewModel>()
```

Comment:
Bluetooth (Home) view model does a lot more and it's associated with the home screen. I would create a new component, `MessageHandler` and extract the logic that it is currently happening in the bluetooth view model into that component. That component can then be injected in the `MainActivityViewModel` and here we can invoke a new action that will internally invoke the handler